### PR TITLE
Add identity column doc

### PR DESCRIPTION
**Pull request by zhipengmao-db**
#### Which Delta project/connector is this regarding?

- [x] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description
It adds doc for identity column.

## How was this patch tested?
Doc change.

## Does this PR introduce _any_ user-facing changes?
No.

### CODE REVIEW

**Review by c27kwan — COMMENTED** [review]
Good work! Can you add identity columns to the table in https://docs.delta.io/latest/versioning.html#what-delta-lake-features-require-client-upgrades?
and here https://docs.delta.io/latest/versioning.html#features-by-protocol-version

Even though identity columns cannot be created with SQL syntax yet, you can still use CTAS on an already created table. Can you add some docs on that?

**Review by c27kwan — APPROVED** [review]
lgtm!

File: docs/source/delta-batch.md [hunk 1]
```diff
@@ -407,6 +407,150 @@ You can use an [EXPLAIN](https://spark.apache.org/docs/latest/sql-ref-syntax-qry
...
+You can use `ALTER TABLE table_name ALTER COLUMN column_name SYNC IDENTITY` to synchronize the metadata of an identity column with the actual data. When you write your own values to an identity column, it might not comply with the metadata. This option evaluates the state and updates the metadata to be consistent with the actual data. After this command, the next automatically assigned identity value will start from `start + (n + 1) * step`, where `n` is the smallest value that satisfies `start + n * step >= max()` (for a positive step).
+
+#### CTAS and identity columns
+
+You cannot define schema, identity column constraints, or any other table specifications when using a `CREATE TABLE table_name AS SELECT` (CTAS) statement.
+
+To create a new table with an identity column and populate it with existing data, do the following:
+
+#. Create a table with the correct schema, including the identity column definition and other table properties.
+#. Run an insertion operation.
```
Comment (allisonport-db) [inline docs/source/delta-batch.md:477]: What does `#.` look like? have you built this page and looked at the html?

File: docs/source/delta-batch.md [hunk 2]
```diff
@@ -407,6 +407,150 @@ You can use an [EXPLAIN](https://spark.apache.org/docs/latest/sql-ref-syntax-qry
 
 <a id="special-chars-in-col-name"></a>
 
+### Use identity columns
+.. important:: Declaring an identity column on a Delta table disables concurrent transactions. Only use identity columns in use cases where concurrent writes to the target table are not required.
+
+<Delta> identity columns are a type of generated column that assigns unique values for each record inserted into a table. The following example shows how to declare an identity column during a create table command:
```
Comment (allisonport-db) [inline docs/source/delta-batch.md:413]: Clarify that it's supported in 3.3 and above please

File: docs/source/versioning.md [hunk 3]
```diff
@@ -28,6 +28,7 @@ The following <Delta> features break forward compatibility. Features are enabled
    Clustering, [Delta Lake 3.1.0](https://github.com/delta-io/delta/releases/tag/v3.1.0),[_](/delta-clustering.md)
    Row Tracking, [Delta Lake 3.2.0](https://github.com/delta-io/delta/releases/tag/v3.2.0),[_](/delta-row-tracking.md)
    Type widening (Preview),[Delta Lake 3.2.0](https://github.com/delta-io/delta/releases/tag/v3.2.0),[_](/delta-type-widening.md)
+   Identity columns, [Delta Lake 3.3.0](https://github.com/delta-io/delta/releases/tag/v3.3.0),[_](/delta-batch.md#use-identity-columns)
```
[the remark this answers is not in the document]
Comment (allisonport-db) [reply docs/source/versioning.md:31]: No this seems fine

File: docs/source/delta-batch.md (hunk 1 above)
Comment (zhipengmao-db) [reply docs/source/delta-batch.md:477]: Screenshot 2024-12-11 at 21 57 15

**Review by allisonport-db — APPROVED** [review]

**Pull request merged**